lib/improver/tests/nowcasting/accumulation/test_Accumulation.py

```diff
@@ -0,0 +1,421 @@
+class Test_sort_cubes_by_time(rate_cube_set_up):
+    """Tests the input cubes are sorted in time ascending order."""
+
+    def test_returns_cubelist(self):
+        """Test function returns a cubelist."""
+
+        result, _ = Accumulation.sort_cubes_by_time(self.cubes)
+        self.assertIsInstance(result, iris.cube.CubeList)
+
+    def test_reorders(self):
+        """Test function reorders a cubelist that is not time ordered."""
+
+        expected = [cube.coord('time').points[0] for cube in self.cubes]
+        self.cubes = self.cubes[::-1]
+        reordered = [cube.coord('time').points[0] for cube in self.cubes]
+
+        result, _ = Accumulation.sort_cubes_by_time(self.cubes)
+        result_times = [cube.coord('time').points[0] for cube in result]
+
+        self.assertIsInstance(result, iris.cube.CubeList)
+        self.assertEqual(result_times, expected)
+        self.assertNotEqual(result_times, reordered)
+
+    def test_times(self):
+        """Test function returns the correct times for the sorted cubes."""
+
+        expected = [1510286400, 1510286460, 1510286520, 1510286580,
+                    1510286640, 1510286700, 1510286760, 1510286820,
+                    1510286880, 1510286940, 1510287000]
+        _, times = Accumulation.sort_cubes_by_time(self.cubes)
+
+        self.assertArrayEqual(times, expected)
+
+
+class Test_get_period_sets(rate_cube_set_up):
+    """Tests the requested accumulation_period can be constructed from the
+    input cubes and return lists of the required inputs to construct each
+    such period."""
+
+    def test_returns_correct_type(self):
+        """Test function returns the expected list."""
+
+        time_interval = 60
+        plugin = Accumulation(accumulation_period=120)
+        result = plugin.get_period_sets(time_interval, self.cubes)
+        self.assertIsInstance(result, list)
+
+    def test_returns_all_cubes_if_period_unspecified(self):
+        """Test function returns a list containing the original cube list if
+        the accumulation_period is not set."""
+
+        time_interval = 60
+        plugin = Accumulation()
+        result = plugin.get_period_sets(time_interval, self.cubes)
+        self.assertSequenceEqual(result, [self.cubes])
+
+    @ManageWarnings(record=True)
+    def test_raises_warning_for_unused_cubes(self, warning_list=None):
+        """Test function raises a warning when there are insufficient cubes to
+        complete the last period."""
+
+        time_interval = 60
+        warning_msg = (
+            "The provided cubes result in a partial period given the specified"
+            " accumulation_period, i.e. the number of cubes is insufficient to"
+            " give a set of complete periods. Only complete periods will be"
+            " returned.")
+
+        expected = [self.cubes[0:4], self.cubes[3:7], self.cubes[6:10]]
+
+        plugin = Accumulation(accumulation_period=180)
+        result = plugin.get_period_sets(time_interval, self.cubes)
```

Comment:
This is trying to construct the list of files for a series of 3 minute accumulations from input fields separated by 1 minute.  So it's looking to split a list of 11 files into subsets of 4 files, each starting at multiples of 3.  It gets 3 complete subsets but then has 2 fields left over that are not used, so it raises a warning about this.

It took me a while to work this out.  Could there be a bit more explanation in the test docstrings to help users interpret this please.

Reply:
Done.